src/logger/logger.class.ts

```diff
@@ -1,20 +1,20 @@
 import { Level } from './level'
 import { Format, FormatOptions } from './format'
-import { createVault, VaultOptions, IVault, Vault } from './vault'
-import { ILogger } from './logger.interface'
+import { createVault, VaultOptions, Vault, VaultKind } from './vault'
+import { Logger } from './logger.interface'
 import { LogContext } from './log-context.class'
 
-export type LoggerOptions = {
+export type LoggerBaseOptions = {
   level: Level
   format: FormatOptions
   vault: VaultOptions | VaultOptions[]
 }
 
-export class Logger implements ILogger {
-  vaults: IVault[]
+export class LoggerBase implements Logger {
```

Comment:
You are probably right. The class serves as the base for the environment loggers – as well as ones defined outside the library.
My best suggestion is to use `GenericLogger` instead, since it is not specialized.

Reply:
The logger doesn't make use of Generics, so having that in the name, might be slightly misleading. 

Naming it `GeneralLogger`, should be a more fitting name in that sense. This is especially true, if it is suppose to not have ´abstract´, thus allowing it to be instantiated as is.